numba/cuda/api.py

```diff
@@ -231,8 +231,32 @@ def _fill_stride_by_order(shape, dtype, order):
 def device_array_like(ary, stream=0):
     """Call cuda.devicearray() with information from the array.
     """
-    return device_array(shape=ary.shape, dtype=ary.dtype,
-                        strides=ary.strides, stream=stream)
+    # Avoid attempting to recompute strides if the default strides will be
+    # sufficient to create a contiguous array.
+    if ary.is_c_contiguous() or ary.ndim <= 1:
```

Comment:
Is being `F`-contiguous insufficient?

Reply:
Looks like it could be - will try and add a check and appropriate behaviour for this.

Reply:
(could be sufficient, that is!)

Reply:
great, thanks for checking.